src/main/java/com/atlassian/bitbucket/jenkins/internal/status/LocalSCMListener.java

```diff
@@ -112,6 +119,23 @@ public void onCheckout(Run<?, ?> build, SCM scm, FilePath workspace, TaskListene
         buildStatusPoster.postBuildStatus(revisionAction, build, listener);
     }
 
+    private void buildEnvironment(GitSCM underlyingScm, Run<?, ?> build, Map<String, String> env) {
+        underlyingScm.buildEnvironment(build, env);
```

Comment:
`GitSCM#buildenvironment` is also setting the branch, to the remoteBranchName, and we're overriding it. Is GitSCM setting it incorrectly?

Reply:
> `GitSCM#buildenvironment` is also setting the branch, to the remoteBranchName, and we're overriding it. Is GitSCM setting it incorrectly?

Yes and no. It sets the `GIT_BRANCH` to `head.getName()` which is correct. HOWEVER, the`BitbucketPullRequestSCMHead` has name to be `PR-1` (or whatever number). And it _has_ to do that as it is also used to display things on the dashboard/name the build etc. If we were to opt for the branch name there you'd have multiple _identical_ builds if you have more than one PR from a branch.